js/editor-controller.js

```diff
@@ -0,0 +1,235 @@
+/**
+ * The Editor can be controlled by remote hosts that wrap the editor with
+ * an iframe or in some kind of webview.
+ */
+'use strict';
+
+/**
+ * @returns An object with the fs wrapper.
+ */
+var EditorController = function() {
+    var _returnObj = {};
+
+    var _editorActions = null;
+    var _controllerHost = null;
+
+    /** Constants used for iframe messaging */
+    var CONTROLLER_MESSAGING = Object.freeze({
+        // Embed editor host type
+        type: 'pyeditor',
+        // Embed editor messaging actions
+        actions: {
+            workspacesync: 'workspacesync',
+            workspacesave: 'workspacesave',
+            workspaceloaded: 'workspaceloaded',
+            importproject: 'importproject',
+            loadhex: 'loadhex',
+            loadfile: 'loadfile',
+            downloadmode: 'downloadmode',
+            savefile: 'save',
+            flashhex: 'download',
```

Comment:
I've added these two messages that go from the editor to the host with the original string as done in pxt and discussed in https://github.com/bbcmicrobit/PythonEditor/pull/368/files#r574749489:
https://github.com/microsoft/pxt/blob/master/pxteditor/editor.ts#L554

If we are not going to use the message format from MakeCode maybe we should replace the strings as well? 

@microbit-sam I assume this is fine as well for the Android app? Does it currently use postMessage with MakeCode?

Reply:
The Android app doesn't use postMessage at all at the moment, so no issues here

Reply:
Oh right, how does it currently work with the MakeCode app? Does it inject custom JS?

Reply:
It takes the hex file encoded as a b64 URI and then decodes that into a hex file
It works, but I expect could break easily with URI length limits

Reply:
@martinwork @microbit-sam it sounds like we can rename the two strings to have the same value as their key:

```suggestion
            savefile: 'savefile',
            flashhex: 'flashhex',
```

Reply:
Yes, those names are better.